### ISSUE TITLE
Scala versions, DL4J/ND4J/DataVec etc modules, 2.12 release

### ISSUE THREAD

**Issue by AlexDBlack** [eclipse/deeplearning4j#8151]
Up to this point, we have released only Scala 2.10 and 2.11 versions of our libraries.
Scala 2.12 is seeing increasing usage (November 2016), and Scala 2.13 is also out now (June 2019).

Adding scala 2.12 is a matter of when, not if. If possible, it would be nice to have Scala 2.12 in the 1.0.0-beta5 release, along with the 2.10 and 2.11, especially given the recent improvements to ND4S.
Even if we don't add 2.12 support for beta5, we'll still need some changes due to the Play framework version upgrade (no 2.10 support).

Our main complication is that certain libraries (and, especially a single version of some libraries) don't support all of Scala 2.10, 2.11 and 2.12.
The end result:
* Our simple "find and replace" change-scala-versions.sh script will probably need some more special cases
    - For example, "Use version X for scala 2.10, use version Y for scala 2.11" etc
* We may need a few version upgrades for some dependencies, to a version that has all of 2.10/2.11/2.12 versions
* In some cases, we will need to exclude whole modules from the build/release for a given Scala version. For example:
    - No Scala 2.12 for DL4J Spark 1.0
    - No Scala 2.10 for DL4J UI (once this is merged: https://github.com/eclipse/deeplearning4j/pull/8098)
* In some other cases, a minor reorganization of the tests or test dependencies may be required (for example, Spark has deeplearning4j-play_2.xx test scope dependency; can't build Spark 1 2.10 if there's no deeplearning4j-play_2.10)

Here's the filtered/processed output of `mvn dependency:list` on the PR above (which has been rebased to master).
https://gist.github.com/AlexDBlack/b0a750f03f2cc8cf79fa30353cd4445c

Script used to process the output: https://gist.github.com/AlexDBlack/2f16f3ee01f159c785f36eadacdec4c4

Related:
https://github.com/eclipse/deeplearning4j/issues/6269
https://github.com/eclipse/deeplearning4j/issues/5259

**Comment by AlexDBlack** [comment]
After talking internally: we have decided to drop Spark 1.x, and also drop scala 2.10
1.0.0-beta5 (and future) releases will be Spark 2.x only, and both Scala 2.11 and 2.12
